conseil-common/src/main/scala/tech/cryptonomic/conseil/common/metadata/MetadataService.scala

```diff
@@ -17,22 +16,24 @@ class MetadataService(
     config: PlatformsConfiguration,
     transformation: UnitTransformation,
     cacheConfiguration: AttributeValuesCacheConfiguration,
-    platformDiscoveryOperations: PlatformDiscoveryOperations
+    platformDiscoveryOperations: Map[String, PlatformDiscoveryOperations]
 )(implicit apiExecutionContext: ExecutionContext) {
 
-  private val platforms = transformation.overridePlatforms(ConfigUtil.getPlatforms(config))
+  private val platforms = transformation.overridePlatforms(config.getPlatforms)
 
   private val networks = platforms.map { platform =>
-    platform.path -> transformation.overrideNetworks(platform.path, ConfigUtil.getNetworks(config, platform.name))
+    platform.path -> transformation.overrideNetworks(platform.path, config.getNetworks(platform.name))
   }.toMap
 
   private val entities = {
-    val allEntities = Await.result(platformDiscoveryOperations.getEntities, 2 second)
+    val allEntities = platforms
+      .map(platform => platform.path -> Await.result(platformDiscoveryOperations(platform.name).getEntities, 2 second))
+      .toMap
 
     networks.values.flatten
       .map(_.path)
       .map(
-        networkPath => networkPath -> transformation.overrideEntities(networkPath, allEntities)
+        networkPath => networkPath -> transformation.overrideEntities(networkPath, allEntities(networkPath.up))
```

Comment:
What would you say of having local methods that extracts the platform from different paths, using explicit names?
```scala
private def platformPath(path: NetworkPath): PlatformPath = path.up
private def platformPath(path: EntityPath): PlatformPath = path.up.up
...
```

Reply:
at call site we would have, e.g.

```scala
if (exists(path)) {
  platformDiscoveryOperations(platformPath(path).platform).getEntities.map {...}
}
```

Reply:
or something like `upToPlatform(path: ...): PlatformPath`, which might be more clear?

Reply:
I'm thinking more about making this: 
```
private def toPlatformName(path: NetworkPath): String = path.up.platform
private def toPlatformName(path: EntityPath): String = toPlatformName(path.up)
```
mostly because there is only 1 (out of 6-7) usages that calls `.platform` directly after it, so I guess that would work better, to call `.platform` directly in this method.

Reply:
Yes, fine for me.
Let me check if I got it: since the actual PlatformPath is seldom used without calling the platform name immediately after, you want to make that operation easier, and do things explicitly for the only case needing the actual path?

If that's what you meant, it's perfectly ok with me

Reply:
Yew, that is correct.